### PR TITLE
provide better error from ansible-galaxy when role version is in conflict

### LINKED ISSUE #52111: ansible-galaxy has cryptic error message when role version is wrong
##### SUMMARY
when specifying the wrong version for a ROLE that is downloading from GIT, ansible-galaxy fails with a cryptic message instead of complaining on missing tag/version 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible-galaxy 

##### ANSIBLE VERSION
```
ansible-galaxy 2.7.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/srv/var/jenkins/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /srv/var/jenkins/.local/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible-galaxy
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

```

##### CONFIGURATION
```
```

##### OS / ENVIRONMENT
Centos 7 

##### STEPS TO REPRODUCE

```
[jenkins@vm61 dummy]$ ansible-galaxy --version
ansible-galaxy 2.7.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/srv/var/jenkins/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /srv/var/jenkins/.local/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible-galaxy
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
[jenkins@vm61 dummy]$ cat roles.yml
# Standards: 0.1
---
- src: ssh://git.myserver.net:29418/operations/ansible/dummy
  scm: git
  version: '0.2'

[jenkins@vm61 dummy]$ ansible-galaxy install -r roles.yml  --force
 [WARNING]: - dummy was NOT installed successfully: - command /bin/git checkout 0.2 failed in directory /srv/var/jenkins/.ansible/tmp/ansible-local-22840YMo_1g/tmpi_69sw/dummy (rc=1)

ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
[jenkins@vm61 dummy]$ 

```

##### EXPECTED RESULTS

ansible-galaxy provides error about missing tag/version 

##### ACTUAL RESULTS
the error bellow is not clear enough on why the checkout failed, can lead to hours of troubleshooting and code reading .... 
```
 [WARNING]: - dummy was NOT installed successfully: - command /bin/git checkout 0.2 failed in directory /srv/var/jenkins/.ansible/tmp/ansible-local-21277UaRVoO/tmp7UCvAY/dummy (rc=1)
```

### PR DESCRIPTION
**Pull request by sudhirmohanraj**
##### SUMMARY
Added the version name to the log generated. Fixed #52111

##### ISSUE TYPE

- Bugfix Pull Request fixes #52111

##### COMPONENT NAME
ansible/cli

##### ADDITIONAL INFORMATION
Added the version name to the log.

### CODE REVIEW

**Comment by sudhirmohanraj** [comment]
@vmindru can you review the fix.

**Comment by mattclay** [comment]
The reported sanity test failures are valid. The integration test failures are unrelated to the changes.

**Comment by vmindru** [comment]
I am not sure this will help anyhow, simply printing the TAG version I have in the config is not much different than the current behavior. I would expect rather an explicit Failure cause e.g. error that the tag is missing remotely.

File: lib/ansible/cli/galaxy.py [hunk 1]
```diff
@@ -391,7 +391,7 @@ def execute_install(self):
             try:
                 installed = role.install()
             except AnsibleError as e:
-                display.warning(u"- %s was NOT installed successfully: %s " % (role.name, to_text(e)))
+                display.warning(u"- %s  and %s version was NOT installed successfully: %s " % (role.name, role.version, to_text(e)))
```
Comment (bcoca) [inline lib/ansible/cli/galaxy.py:394]: remove 'and' and 'version' as the name + version should be good enough to identify and the 'and' is confusion as it seems you installed (or failed to ) 2 things.

**Review by bcoca — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@sudhirmohanraj Are you still working on this, let us know ? Thanks, needs_info

**Pull request closed without merging**